x/gov/client/cli/query.go

```diff
@@ -280,25 +299,32 @@ $ gaiacli query gov deposit 1 cosmos1skjwj5whet0lpe65qaq4rpq03hjxlwd9nf39lk
 				return fmt.Errorf("Failed to fetch proposal-id %d: %s", proposalID, err)
 			}
 
-			// Get the depositer address
 			depositorAddr, err := sdk.AccAddressFromBech32(args[1])
 			if err != nil {
 				return err
 			}
 
-			// Construct query
 			params := gov.NewQueryDepositParams(proposalID, depositorAddr)
 			bz, err := cdc.MarshalJSON(params)
 			if err != nil {
 				return err
 			}
 
-			// Query store
 			res, err := cliCtx.QueryWithData(fmt.Sprintf("custom/%s/deposit", queryRoute), bz)
 			if err != nil {
 				return err
 			}
 
+			var deposit gov.Deposit
+			cdc.UnmarshalJSON(res, &deposit)
+
+			if deposit.Empty() {
+				res, err = gcutils.QueryDepositByTxQuery(cdc, cliCtx, params)
```

Comment:
shouldn't we get the proposal and query the deposit tx if it's not on deposit or voting period ? I mean, shouldn't we have an `if...else` around `res, err := cliCtx.QueryWithData(fmt.Sprintf("custom/%s/deposit", queryRoute)`

Reply:
Hmmm I dont think so. I think its implicit. Deposit will be empty in the case where its not voting or deposit period, no?